src/app/cli/src/commands/ingest_command.rs

```diff
@@ -149,3 +188,186 @@ impl Command for IngestCommand {
         Ok(())
     }
 }
+
+///////////////////////////////////////////////////////////////////////////////
+
+pub(crate) struct PushIngestProgress {
+    dataset_handle: DatasetHandle,
+    multi_progress: Arc<indicatif::MultiProgress>,
+    state: Mutex<PushIngestProgressState>,
+}
+
+struct PushIngestProgressState {
+    stage: PushIngestStage,
+    curr_progress: indicatif::ProgressBar,
+}
+
+impl PushIngestProgress {
+    pub fn new(
+        dataset_handle: &DatasetHandle,
+        multi_progress: Arc<indicatif::MultiProgress>,
+    ) -> Self {
+        Self {
+            state: Mutex::new(PushIngestProgressState {
+                stage: PushIngestStage::CheckSource,
+                curr_progress: multi_progress.add(Self::new_spinner(&Self::spinner_message(
+                    dataset_handle,
+                    0,
+                    "Checking source for updates",
+                ))),
+            }),
+            dataset_handle: dataset_handle.clone(),
+            multi_progress,
+        }
+    }
+
+    fn new_spinner(msg: &str) -> indicatif::ProgressBar {
+        let pb = indicatif::ProgressBar::hidden();
+        let style = indicatif::ProgressStyle::default_spinner()
+            .template("{spinner:.cyan} {msg}")
+            .unwrap();
+        pb.set_style(style);
+        pb.set_message(msg.to_owned());
+        pb.enable_steady_tick(Duration::from_millis(100));
+        pb
+    }
+
+    fn spinner_message<T: std::fmt::Display>(
+        dataset_handle: &DatasetHandle,
+        step: u32,
+        msg: T,
+    ) -> String {
+        let step_str = format!("[{}/7]", step + 1);
+        let dataset = format!("({})", dataset_handle.alias);
+        format!(
+            "{} {} {}",
+            console::style(step_str).bold().dim(),
+            msg,
+            console::style(dataset).dim(),
+        )
+    }
+
+    fn message_for_stage(&self, stage: PushIngestStage) -> String {
+        let msg = match stage {
+            PushIngestStage::CheckSource => "Checking source for updates",
+            PushIngestStage::Fetch => "Downloading data",
```

Comment:
Didn't expect to see such phases for Push flow

Reply:
I'm still not fully sure about the scope of responsibilities of `PushIngestService`:
- It could accept just an input `DataFrame` and only handle preprocessing/merge/commit
- It could (as it is now) also handle reading raw data, and in this case:
  - some data can be read directly into a `DataFrame` (if location and data format are natively supported by DataFusion)
  - some data has to be fetched/transformed before reading if the location or format are not natively supported by DataFusion (e.g. if in future we will have push data arriving via Kafka)

It's because of the latter I left the `CheckSource` and `Fetch` stages.

I will remove them for now, not to muddy the current state of things, until we get to the Kafka stage. Perhaps Kafka micro-batching will be handled by a completely different service.